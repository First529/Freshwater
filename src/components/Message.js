export class Message {
  constructor(sender, destination, text, time) {
    this.sender = sender;
    this.destination = destination;
    this.message = text;
    this.time = time;
  }
}
