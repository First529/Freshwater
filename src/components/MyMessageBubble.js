import React from "react";

export default class MessageBubble extends React.Component {
  render() {
    return (
      <div className="message">
        <div className="message-text-me">{this.props.text}</div>
        <div className="message-time-me">{this.props.time}</div>
      </div>
    );
  }
}
