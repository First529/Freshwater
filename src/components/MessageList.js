import React from "react";
import MessageBubble from "./MessageBubble";
import MyMessageBubble from "./MyMessageBubble";

export default class MessageList extends React.Component {
  constructor() {
    super();
    this.date = new Date().toLocaleDateString()
    this.state = {
      username: "me"
    };
  }
  render() {
    return (
      <div className="message-list">
        <div className="date">
          <div className="date-text">  {this.date} </div>
        </div>
        {this.props.messages.map((message, index) => {
          if (message.sender === this.state.username) {
            return (
              <div className="my-message-slot">
                <MyMessageBubble text={message.text} time={message.time} />
              </div>
            );
          } else {
            return (
              <div className="message-slot">
                <div className="message-sender">{message.sender}</div>
                <MessageBubble text={message.text} time={message.time} />
              </div>
            );
          }
        })}
      </div>
    );
  }
}
