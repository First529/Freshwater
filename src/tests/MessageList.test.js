import React from "react";
import { shallow } from "enzyme";
import MessageList from "../components/MessageList";
import Message from "../components/Message";
import MyMessageBubble from "../components/MyMessageBubble";

const TEMP = [
  new Message(
    "test",
    "Public",
    "Hello",
    "15:30",
    "other",
    "Public",
    "Hello",
    "15:30"
  )
];

describe("SendMessage", () => {
  it("renders correctly", () => {
    const component = shallow(<MessageList messages={TEMP} />);
    expect(component).toMatchSnapshot();
  });

  it("render user on the correct side", () => {
    const component = shallow(<MessageList messages={TEMP} />, {
      disableLifecycleMethods: true
    });
    component.setState({ username: "test" });
    expect(component.render(<MyMessageBubble />));
  });
});
