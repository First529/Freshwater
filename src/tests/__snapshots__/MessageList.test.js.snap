// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendMessage renders correctly 1`] = `
<div
  className="message-list"
>
  <div
    className="date"
  >
    <div
      className="date-text"
    >
       
       
    </div>
  </div>
  <div
    className="message-slot"
    key="0"
  >
    <div
      className="message-sender"
    >
      test
    </div>
    <MessageBubble
      message="Hello"
      time="15:30"
    />
  </div>
</div>
`;
